### ISSUE TITLE
Regression: basic authentication in URL does not work

### ISSUE THREAD

**Issue by dottedmag** [rss2email/rss2email#84]
```
% r2e list  | grep netch
23: [*] netch (https://dottedmag:<redacted password>@netch.dreamwidth.org/data/atom?auth=digest -> dottedmag@dottedmag.net)
% r2e run --no-send
processing error: expected bytes-like object, not str: netch (https://dottedmag:<redacted password>@netch.dreamwidth.org/data/atom?auth=digest -> dottedmag@dottedmag.net)
% r2e --version
rss2email 3.9
%
```

It used to work in an ancient Python 2 version of `rss2email`.

**Comment by Ekleog** [comment]
Are there non-ascii characters in the `<redacted password>`? That sounds like something that could cause this issue, and be due to the 2->3 switch.

**Comment by dottedmag** [comment]
Nope, purely uppercase and lowercase ASCII Latin and digits.

**Comment by Profpatsch** [comment]
That’s a classic issue; we are probably using a bytes-based function and passing a unicode string.

**Comment by lkmorlan** [comment]
URLs of this format used to work:
https://user:pass@example.com/path

Now I get "processing error: nonnumeric port: 'pass'". It is not correctly parsing the URL and interprets the password to be the port.

**Comment by Ekleog** [comment]
With what version of feedparser is this new error? Also, do you remember with what versions of rss2email/feedparser the “previously working” behavior was?

**Comment by lkmorlan** [comment]
I think it was a 2.x version of rss2email. I copied that comment from [my 2017 report in the wking repository](/wking/rss2email/issues/80#issuecomment-692035828).

**Comment by Ekleog** [comment]
Hmm ok, thank you :) just, do you know whether this is happening with feedparser 5 or 6 ?

**Comment by lkmorlan** [comment]
I am seeing this problem in the rss2email package in Debian which depends on python3-feedparser 5.2.

**Comment by auouymous** [comment]
http://test:password@foo.com/bar.php can be used to test this since nothing is sent to the server.

```
[ERROR] processing error: a bytes-like object is required, not 'str': test (http://test:password@foo.com/bar.php -> rss2email)
```

The problem is entirely in feedparser. Encoding user_passwd (in feedparser) fixes the initial error and then throws the same error in `return opener.open(request)`. That code is different in 6.0 so it might have been fixed.

```
-    auth = base64.standard_b64encode(user_passwd).strip()
+    auth = base64.standard_b64encode(user_passwd.encode()).strip()
```

**Comment by Ekleog** [comment]
Hmm ok, so I guess let's block this on #73 and see whether it still happens after that is fixed, and then reevaluate whether it's an upstream issue or a local one :)

**Comment by auouymous** [comment]
Feedparser 6.0.1 has the same problem, https://github.com/kurtmckee/feedparser/pull/234 fixes it.

**Comment by auouymous** [comment]
This is fixed with feedparser 6.0.5. After #188 is merged the required version in setup.py and pyproject.toml should be bumped.

**Comment by Ekleog** [comment]
Should hopefully be fixed now :) Thank you all for your help!
